Userland/Applications/TextEditor/MainWidget.cpp

```diff
@@ -667,7 +667,15 @@ bool MainWidget::open_file(const String& path)
 
     m_editor->set_text(file->read_all());
 
-    set_path(path);
+    if (!path.starts_with("/"sv)) {
+        char* buf = getcwd(nullptr, 0);
+        auto lexical_path = LexicalPath::join(buf, path);
+        free(buf);
+
+        set_path(lexical_path.string());
+    } else {
+        set_path(path);
+    }
```

Comment:
Can we use `Core::File::real_path_for()` to achieve all this instead? :)

Reply:
From my understanding we cannot because `realpath` requires unveiled access to the path.

I'm working around this by ensuring FileSystemAccessServer only returns absolute paths, and that a filename passed as an argument will be resolve before hand.

Reply:
Removed this change and have replaced it with `VERIFY(path.starts_with("/"sv));` at the top of `MainWidget::open_file`, along with the ideas above.